src/Lib/Lionk.Core/Model/Component/Cyclic/ICyclicComponent.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright © 2024 Lionk Project
+
+namespace Lionk.Core.Component;
+
+/// <summary>
+/// This interface defines a cyclic component.
+/// </summary>
+public interface ICyclicComponent : IExecutableComponent
+{
+    /// <summary>
+    /// Gets the execution frequency of the component.
+    /// </summary>
+    TimeSpan ExecutionFrequency { get; }
+
+    /// <summary>
+    /// Gets the last execution time of the component.
+    /// </summary>
+    DateTime LastExecution { get; }
+
+    DateTime NextExecution => StartedDate.AddMilliseconds(ExecutionFrequency.TotalMilliseconds * NbCycle);
```

Comment:
Should not be LastExecution + ExecutionFrequency ?

Reply:
No, if you start at 00:00 with a frequency of 5 seconds, the next run will take place at 00:05 and the next run at 00:10.
If you use the last run, you start at 00:00 and have the next run at 00:05. If the execution takes place at 00:06, the next execution takes place at 00:11.

With the solution we've implemented, we can make up for lost time by respecting a multiple of the frequency.

The LastExecution is just an information. Now it setted inside the running cycle

Reply:
Define an option to choose between planned run or chained.